lib/significant-other.js

```diff
@@ -7,35 +7,55 @@ import {
 
 export default {
   subscriptions: null,
+  cache: {},
   activate(state) {
     // Events subscribed to in atom's system can be easily cleaned up with a
     // CompositeDisposable
-    this.subscriptions = new CompositeDisposable();
+    this.subscriptions = new CompositeDisposable()
+    this.cache = state.cache || {}
 
     // Register command that toggles this view
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
-      'significant-other:toggle': () => this.toggle()
+      'significant-other:toggle': () => this.toggle(),
+      'significant-other:clear-cache': () => this.clearCache()
     }));
   },
 
   deactivate() {
-    this.subscriptions.dispose();
+    this.subscriptions.dispose()
+    this.cache = {}
   },
 
   serialize() {
-    return {}
+    return {
+      cache: this.cache
+    }
   },
 
   getFilePath() {
     return atom.workspace.getActiveTextEditor().getPath()
   },
 
+  clearCache() {
+    this.cache = {}
+  },
+
   toggle() {
     const filePath = this.getFilePath()
     const matchmaker = Matchmaker.for(filePath)
 
+    // use cached path if we can
+    if (this.cache[filePath]) {
+      atom.workspace.open(this.cache[filePath])
```

Comment:
If the cached file path no longer exists on disk (i.e., the file has been deleted since the last time significant-other was triggered), this will open a blank tab. Instead of letting that happen, I think it would be helpful to verify that the cached file path still exists on disk:

- If the cached file path still exists on disk, then open the file.
- Otherwise, use `Matchmaker` to look for a match.

[`existsSync`](https://github.com/atom/fs-plus#existssyncpathtocheck) would let us determine whether the cached file path still exists.

@shemerey: Would you be willing to make that change?